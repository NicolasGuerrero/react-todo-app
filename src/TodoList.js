import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo'
import { v4 as uuid } from 'uuid';

function TodoList() {
  const [todos, setTodos] = useState([]);

  //render Todo items
  const renderTodoList = () => {
    return (
      <div>
        {todos.map(todo => (
          <Todo key={todo.id}
            id={todo.id}
            task={todo.task}
            removeTodo={removeTodo}
          />
        ))}
      </div>
    );
  };
  // add Todo item
  const addTodo = todo => {
    let newTodo = { ...todo, id: uuid() };
    setTodos(todos => [...todos, newTodo]);
  }

  // remove Todo item
  const removeTodo = id => {
    setTodos(todos.filter(todo => todo.id !== id));
  }

  // return 
  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      {renderTodoList()}
    </div>
  )
};


export default TodoList;