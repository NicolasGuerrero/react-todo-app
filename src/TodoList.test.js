import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, task = "Buy Groceries") {
  const taskInput = todoList.getByLabelText("Task:");

  fireEvent.change(taskInput, { target: { value: task}});

  const button = todoList.getByText("Add new task");
  fireEvent.click(button);
}

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot - new TodoList", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("it can add a new todo item", function() {
  const todoList = render(<TodoList />);

  expect(todoList.queryByText("X")).not.toBeInTheDocument();

  addTodo(todoList);

  expect(todoList.queryByText("X")).toBeInTheDocument();
});

it("it can delete a box", function() {
  const todoList = render(<TodoList />);

  addTodo(todoList);

  expect(todoList.queryByText("X")).toBeInTheDocument();

  const removeButton = todoList.queryByText("X");

  fireEvent.click(removeButton);

  expect(todoList.queryByText("X")).not.toBeInTheDocument();
});
