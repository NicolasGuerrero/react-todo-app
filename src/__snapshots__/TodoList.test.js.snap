// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - new TodoList 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="container"
    >
      <form
        class="NewTodoForm"
      >
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          value=""
        />
        <button
          class="NewTodoForm-button"
        >
          Add new task
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
